### PR TITLE
fix: historical front-run

### PR DESCRIPTION
**Pull request by eddort**

### CODE REVIEW

File: src/guardian/staking-module-guard/staking-module-guard.service.ts [hunk 1]
```diff
@@ -111,6 +111,100 @@ export class StakingModuleGuardService {
...
+
+    const frontRunnedDepositEvents = duplicatedDepositEvents.filter(
+      (suspectedEvent) => {
+        // get event from lido map
+        const sameKeyLidoDeposit =
+          potentialLidoDepositsKeysMap[suspectedEvent.pubkey];
+
+        if (!sameKeyLidoDeposit) throw new Error('expected event not found');
+
+        return this.isEarlyDeposit(suspectedEvent, sameKeyLidoDeposit);
```
Comment (TheDZhon) [inline src/guardian/staking-module-guard/staking-module-guard.service.ts:180]: there is an attack still:

> invalid deposit with Lido wc + pubkey ⇒ frontrunned deposit with malicious wc + Lido pubkey ⇒ valid deposit with Lido wc + pubkey

Comment (TheDZhon) [reply src/guardian/staking-module-guard/staking-module-guard.service.ts:180]: bc here the invalid event might be stored as a reference:
image

**Review by TheDZhon — CHANGES_REQUESTED** [review]
:eyes:

File: src/guardian/staking-module-guard/staking-module-guard.service.ts [hunk 2]
```diff
@@ -111,6 +111,99 @@ export class StakingModuleGuardService {
...
+    if (isSameBlock) {
+      isEarlyDeposit = firstEvent?.logIndex < secondEvent.logIndex;
+    } else {
+      isEarlyDeposit = firstEvent?.blockNumber < secondEvent.blockNumber;
+    }
+
+    return isEarlyDeposit;
+  }
+  /**
+   * Method is not taking into account WC rotation since historical deposits was checked manually
```
[the remark this answers is not in the document]
Comment (eddort) [reply src/guardian/staking-module-guard/staking-module-guard.service.ts]: fixed

File: src/guardian/staking-module-guard/staking-module-guard.service.ts [hunk 3]
```diff
@@ -111,6 +111,99 @@ export class StakingModuleGuardService {
     );
   }
 
+  isEarlyDeposit(
```
[the remark this answers is not in the document]
Comment (eddort) [reply src/guardian/staking-module-guard/staking-module-guard.service.ts]: fixed

File: src/guardian/staking-module-guard/staking-module-guard.service.ts (hunk 1 above)
Comment (eddort) [reply src/guardian/staking-module-guard/staking-module-guard.service.ts]: fixed

**Review by TheDZhon — APPROVED** [review]
LGTM :+1:

**Review by Amuhar — APPROVED** [review]

**Pull request merged**